app/assets/javascripts/mpa/processors/peptide/GOPeptideProcessor.ts

```diff
@@ -0,0 +1,40 @@
+import {ProcessedPeptideContainer} from '../../ProcessedPeptideContainer';
+import { Count } from '../../counts/CountTable';
+import { GOCountTable } from '../../counts/GOCountTable';
+import { stringLiteral } from '@babel/types';
+
+export namespace GOPeptideProcessor
+{
+    export function process(processedPeptides: ProcessedPeptideContainer): GOCountTable 
+    {
+        var peptideCounts = processedPeptides.countTable;
+        var pept2dataResponse = processedPeptides.response;
+
+        var goCounts = new Map<string, Count>();
+        var go2peptide = new Map<string, Set<string>>();
+        var peptide2go = new Map<string, string[]>();
+
+        pept2dataResponse.response.forEach((data, peptide, _) => 
+        {
+            let fas = data.fa.data || [];
+            let peptideCount = peptideCounts.get(peptide)
+
+            Object.keys(fas)
+                .filter(term => term.startsWith("GO:"))
+                .forEach(term => 
+                    {
+                        goCounts.set(term, (goCounts.get(term) || 0) + peptideCount)
+                        
+                        if(!peptide2go.has(peptide))
```

Comment:
Minor remark: I'm not a big fan of if-statements without the corresponding `{` and `}` brackets.